qiskit/synthesis/arithmetic/adders/rv_ripple_carry_adder.py

```diff
@@ -0,0 +1,157 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Compute the sum of two qubit registers without any ancillary qubits."""
+
+from __future__ import annotations
+from math import ceil
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit import QuantumRegister
+
+
+def _mcx_ladder(n_mcx: int, alpha: int):
+    r"""Implements a log-depth MCX ladder circuit as outlined in Algorithm 2 of [1]. The circuit
+    relies on log-depth decomposition of MCX gate that uses conditionally clean ancillae of [2].
+    Selecting :math:`\alpha=1` creates a CX ladder as shown in Fig. 2 of [1] and selecting
+    :math:`\alpha=2` creates a Toffoli ladder as shown in Fig. 3 of [1].
+
+    Args:
+        n_mcx: Number of MCX gates in the ladder.
+        alpha: Number of controls per MCX gate.
+
+    Returns:
+        QuantumCircuit: The MCX ladder circuit.
+
+    References:
+        1. Remaud and Vandaele, Ancilla-free Quantum Adder with Sublinear Depth, 2025.
+        `arXiv:2501.16802 <https://arxiv.org/abs/2501.16802>`__
+
+        2. Khattar and Gidney, Rise of conditionally clean ancillae for optimizing quantum circuits
+        `arXiv:2407.17966 <https://arxiv.org/abs/2407.17966>`__
+    """
+
+    # pylint: disable=expression-not-assigned
+    def helper(qubit_indices, alphas):
```

Comment:
Really minor and not really needed, so up to you.

```suggestion
    def helper(qubit_indices, alphas) -> QuantumCircuit:
```

Reply:
Sure, added a type hint. It isn't a QuantumCircuit, though.